### PR TITLE
[NASA] Move nasa-specific sensors into python config

### PR DESCRIPTION
**Pull request by atanasenko**

### CODE REVIEW

**Comment by lanwin** [comment]
Ah ok now I understand what you mean. That really makes sens. But outdoor temp should stay a real property cause we can grab that from NonNASA too.

**Comment by atanasenko** [comment]
Ah, outdoor is available from non-nasa? I missed that.

Theoretically, we can mimic nasa from non-nasa in this case. 
Instead of
```
target->set_outdoor_temperature(nonpacket_.src, nonpacket_....);
```
we could
```
target->set_custom_sensor(source, MessageNumber::VAR_out_sensor_airout, (float) nonpacket_....);
```
It does simplify setting up supported sensors, and for non-nasa only a couple lines need to be added for such sensors.

**Comment by lanwin** [comment]
Well not sure about that. I would keep it separate for now and check if we could unify that later.

**Comment by atanasenko** [comment]
@lanwin pleas re-review

File: components/samsung_ac/protocol_nasa.cpp [hunk 1]
```diff
@@ -623,14 +622,12 @@ namespace esphome
...
-                double temp = (double)message.value / (double)10;
-                ESP_LOGW(TAG, "s:%s d:%s VAR_in_temp_water_tank_f %f", source.c_str(), dest.c_str(), temp);
-                target->set_water_temperature(source, temp);
+                ESP_LOGW(TAG, "s:%s d:%s VAR_in_temp_water_tank_f %f", source.c_str(), dest.c_str(), message.value);
                 return;
             }
             case MessageNumber::VAR_out_sensor_airout:
             {
-                double temp = (double)message.value / (double)10;
+                double temp = (double) ((int16_t)message.value) / (double)10;
```
Comment (atanasenko) [inline components/samsung_ac/protocol_nasa.cpp:630]: I also slipped a fix to negative temperatures in.

**Pull request merged**